src/app/features/registration/registration.component.html

```diff
@@ -1 +1,130 @@
-<p>registration works!</p>
+<section>
+  <form [formGroup]="form" (ngSubmit)="onSubmit()">
+    <h2>Registration</h2>
+
+    <!-- ======================================================== -->
+    <div class="form-control">
+      <label for="login">Name: </label>
+      <input
+        class="border"
+        id="name"
+        type="text"
+        formControlName="name"
+        [ngClass]="{
+          invalid: form.get('name').invalid && form.get('name').touched
+        }"
+        [ngClass]="{
+          valid: form.get('name').valid && form.get('name').touched
+        }"
+      />
+      <span
+        class="error"
+        *ngIf="form.get('name').invalid && form.get('name').touched"
+      >
+        <small *ngIf="form.get('name').errors['required']">
+          Name is required
+        </small>
+
+        <small
+          *ngIf="
+            form.get('name').errors['minlength'] &&
+            form.get('name').errors['minlength']['requiredLength']
+          "
+          class="error"
+        >
+          Name should be bigger then
+          {{ form.get("name").errors["minlength"]["requiredLength"] }}
+          symbols. Now
+          {{ form.get("name").errors["minlength"]["actualLength"] }}
+          symbols.
+        </small>
+      </span>
+    </div>
+
+    <!-- ======================================================== -->
+    <div class="form-control">
+      <label for="email">Email: </label>
+      <input
+        class="border"
+        id="email"
+        type="email"
+        formControlName="email"
+        placeholder="example@gmail.com"
+        [ngClass]="{
+          invalid: form.get('email').invalid && form.get('email').touched
+        }"
+        [ngClass]="{
+          valid: form.get('email').valid && form.get('email').touched
+        }"
+        appEmailValidator
```

Comment:
we use as a directive in template-driven form, in reactive form, we usually set validators in form settings